models/DataObject/ClassDefinition/Data/Extension/Text.php

```diff
@@ -36,7 +36,7 @@ public function checkValidity($data, $omitMandatoryCheck = false, $params = [])
      */
     public function isEmpty($data)
     {
-        return strlen($data) < 1;
+        return !is_string($data) || strlen($data) < 1;
```

Comment:
```suggestion
        return strlen((string) $data) < 1;
```
I'd go for a cast here instead, which covers numeric data as well.